### PR TITLE
Add aws-machine-controllers

### PR DESCRIPTION
**Pull request by yazug**
from cluster-api-provide-aws

### CODE REVIEW

File: build_amc.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+set -xe
+
+source common.sh
+source build_utils.sh
+
+AMC_VERSION="$1"
+if [ -z "${AMC_VERSION}" ]; then
+    echo "usage: $0 <aws-machine-controller version>" >&2
```
Comment (russellb) [inline build_amc.sh]: s/aws-machine-controller/aws-machine-controllers/

File: build_amc.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    echo "example: $0 4.3.0.ipv6-2019-12-13-0001" >&2
+    exit 1
+fi
+
+check_prereqs $AMC_STREAM || exit 1
+
+build_image "cluster-network-operator" $AMC_VERSION $AMC_STREAM \
```
Comment (russellb) [inline build_amc.sh]: change "cluster-network-operator" here

File: config.sh [hunk 3]
```diff
@@ -115,3 +115,9 @@ OVNKUBE_STREAM=ovn-kubernetes
 OVNKUBE_GIT_URI=https://github.com/openshift-kni/ovn-kubernetes.git
 OVNKUBE_GIT_REF=4.3-ipv6
 OVNKUBE_DOCKERFILE=Dockerfile
+
+#aws-machine-controllers build config
+AMC_STREAM=aws-machine-controllers
+AMC_GIT_URI=https://github.com/openshift-kni/cluster-api-provider-aws
+AMC_GIT_REF=4.3-ipv6
+AMC_DOCKERFILE=Dockerfile
```
Comment (russellb) [inline config.sh:123]: when we create each image here, it's good to check in the openshift/release repo to determine which Dockerfile is used to build the image if there's more than one Dockerfile in the repo.  Just FYI

**Review by russellb — CHANGES_REQUESTED** [review]

File: config.sh (hunk 3 above)
Comment (russellb) [reply config.sh:123]: `Dockerfile` is correct in this case, see https://github.com/openshift/release/blob/master/ci-operator/config/openshift/cluster-api-provider-aws/openshift-cluster-api-provider-aws-release-4.3.yaml#L13-L16

Comment (yazug) [reply config.sh:123]: Thanks found the linkage for the name and confirmed the Dockerfile name

https://github.com/openshift/release/blob/master/ci-operator/config/openshift/cluster-api-provider-aws/openshift-cluster-api-provider-aws-release-4.3.yaml#L14

File: build_amc.sh (hunk 2 above)
Comment (yazug) [reply build_amc.sh]: done

File: build_amc.sh (hunk 1 above)
Comment (yazug) [reply build_amc.sh]: done

**Pull request merged**